src/main/kotlin/cacophony/codegen/instructions/cacophonyInstructions/ControlflowInstructions.kt

```diff
@@ -0,0 +1,90 @@
+package cacophony.codegen.instructions.cacophonyInstructions
+
+import cacophony.codegen.BlockLabel
+import cacophony.codegen.instructions.Instruction
+import cacophony.controlflow.HardwareRegisterMapping
+import cacophony.controlflow.Register
+
+data class PushReg(
+    val reg: Register,
+) : Instruction {
+    override val registersRead: Set<Register> = setOf(reg)
+    override val registersWritten: Set<Register> = setOf()
```

Comment:
As discussed, I'd say that `rsp` should be included in `registersWritten`, and probably also `registersRead` since `push reg` is basically
```
sub rsp, 8
mov [rsp], reg
```